framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+            if (float(decayLine[0]) == float(matchedTimeSteps[timeStepIndex])):
+              for i in xrange (0,len(self.isotopeListMrtau)):
+                if int(self.perturbationNumber) == 1:
+                  self.numDensityLabelListMrtau.append('numDensity'+'|'+self.isotopeListMrtau[i])
+                  self.decayLabelListMrtau.append('decay'+'|'+self.isotopeListMrtau[i])
+                decayListMrtau.append(decayLine[i+1])
+          if breakFlag == 1:
+            break
+    return decayListMrtau
+  
+  def getRRlist(self,instantDict):
+    """
+      Puts the reaction rate labels and the reaction rate values in a list.
+      @ In, instantDict, dictionary
+      @ Out, rrNames, list, list of parameter labels that are in the reaction rate matrix
+      @ Out, rrValues, list, values relative to the labels, within the reaction rate matrix
+    """
+    rrNames = []
+    rrValues = []
+    if self.printSpatialRR is True:
+      for i in xrange(0,len(self.paramList)):
+        for j in xrange(1,int(self.Ngroups) + 1):
+          for k in xrange(1, int(self.Nregions) + 1):
+            if self.paramList[i] == 'Group':
+              pass
+            else:
+              rrNames.append(self.paramList[i]+'|gr'+str(j)+'|reg'+str(k))
+              rrValues.append(instantDict.get('reactionRateInfo').get(str(j)).get(str(k)).get(self.paramList[i]))
+    if self.printSpatialRR is False:
+      for i in xrange(0,len(self.paramList)):
+        if self.paramList[i]   == 'Group':
+          pass
+        elif self.paramList[i] == 'Region':
+          pass
+        else:
+          rrNames.append(self.paramList[i]+'|Total')
+          rrValues.append(instantDict.get('reactionRateInfo').get(self.paramList[i]))
+    if 'Group' in rrNames:
+      rrNames.remove('Group')
+    return rrNames, rrValues
+    
+  def writeCSV(self,instantDict,timeStepIndex,matchedTimeSteps,jobTitle):
+    """
+      Prints the INSTANT/MRTAU data in csv files.
+      @ In, InstantDict, dictionary, contains all the values collected from INSTANT output
+      @ In, timeStepIndex, integer, timestep number
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ In, jobTitle, string, job title parsed from INSTANT input
+      @ Out, None
+      
+    """
+    if self.paramList != []:
+      rrNames, rrValues = self.getRRlist(instantDict)
+      csvOutput = os.path.join(instantDict.get('workingDir'),jobTitle+'-'+self.perturbationNumber+'.csv')
+      if self.phisicsRelap is False:
+        with open(csvOutput, 'a+') as f:
+          instantWriter = csv.writer(f, delimiter=',',quotechar=',', quoting=csv.QUOTE_MINIMAL)
+          if timeStepIndex == 0:
+            instantWriter.writerow(['timeMrTau'] + ['keff'] + ['errorKeff'] + rrNames + instantDict.get('fluxLabelList') + instantDict.get('matFluxLabelList') + instantDict.get('depLabelList') + instantDict.get('decayLabelList') + instantDict.get('xsLabelList') + ['cpuTime'])
```

Comment:
this format is not explained in the manual anywhere

Reply:
hum okay. You want me to list all the possible variable template in the existing_interface, right?

Reply:
Yes